deltametrics/section.py

```diff
@@ -518,10 +703,268 @@ def __init__(self, x=-1):
+
+        The `origin` attempts to detect the land width from bed elevation
+        changes, but should use the value of ``L0`` recorded in the netcdf
+        file, or defined in the cube.
+
+    Parameters
+    ----------
+    *args : :obj:`DataCube` or `StratigraphyCube`
+        The `Cube` object to link for underlying data. This option should be
+        ommitted if using the :obj:`register_section` method of a `Cube`.
+
+    radius : :obj:`float`, `int`, optional
+        The `radius` of the section. This is the distance to locate the
+        section from the :obj:`origin`. If no value is given, the `radius`
+        defaults to half of the minimum model domain edge length if it can be
+        determined, otherwise defaults to ``1``.
+
+    origin : :obj:`tuple` or `list` of `int`, optional
+        The `origin` of the circular section. This is the center of the
+        circle. If no value is given, the origin defaults to the center of the
+        x-direction of the model domain, and offsets into the domain a
+        distance of ``y == L0``, if these values can be determined. I.e., the
+        origin defaults to be centered over the channel inlet. If no value is
+        given, and these values cannot be determined, the origin defaults to
+        ``(0, 0)``.
+
+    **kwargs
+        Keyword arguments are passed to `BaseSection.__init__()`. Supported
+        options are `name`.
+
+    Returns
+    -------
+    section : :obj:`CircularSection`
+        `CircularSection` object with specified parameters. The section is
+        automatically connected to the underlying `Cube` data source if the
+        :obj:`register_section` method of a `Cube` is used to set up the
+        section, or the `Cube` is passed as the first positional argument
+        during instantiation.
+
+    Examples
+    --------
+
+    To create a `CircularSection` that is registered to a `DataCube` with
+    radius ``=30``, and using the default `origin` options:
+
+    .. plot::
+        :include-source:
+
+        >>> rcm8cube = dm.sample_data.cube.rcm8()
+        >>> rcm8cube.register_section('circular', dm.section.CircularSection(radius=30))
+        >>>
+        >>> # show the location and the "velocity" variable
+        >>> fig, ax = plt.subplots(2, 1, figsize=(8, 4))
+        >>> rcm8cube.show_plan('eta', t=-1, ax=ax[0], ticks=True)
+        >>> rcm8cube.sections['circular'].show_trace('r--', ax=ax[0])
+        >>> rcm8cube.sections['circular'].show('velocity', ax=ax[1])
+        >>> plt.show()
+    """
+
+    def __init__(self, *args, radius=None, origin=None, **kwargs):
+
+        self._input_radius = radius
+        self._input_origin = origin
+        super().__init__('circular', *args, **kwargs)
+
+    def _compute_section_coords(self):
+        if (self._input_radius is None):
+            if not (self.cube is None):
+                self.radius = int(np.min(self.cube.shape[1:]) / 2)
+            else:
+                self.radius = 1
```

Comment:
Just out of curiosity, did you have any scenarios in mind where this condition might occur?

Reply:
When I was writing it, I thought it might happen, but then when I was testing, I realized that `_compute_section_coords` is never called until a cube is connected with `.connect()`. So no, I don't think this branch is ever going to be hit. I'm going to remove these branches from all the `Sections`.